addons/account/wizard/account_payment_register.py

```diff
@@ -254,21 +272,34 @@ def _compute_journal_id(self):
                 journal = self.env['account.journal'].search(domain, limit=1)
             wizard.journal_id = journal
 
-    @api.depends('journal_id')
-    def _compute_currency_id(self):
+    @api.depends('company_id', 'can_edit_wizard')
+    def _compute_available_partner_bank_ids(self):
         for wizard in self:
-            wizard.currency_id = wizard.journal_id.currency_id or wizard.source_currency_id or wizard.company_id.currency_id
+            if wizard.can_edit_wizard:
+                batches = wizard._get_batches()
+                bank_partners = batches[0]['lines'].move_id.bank_partner_id
+                wizard.available_partner_bank_ids = bank_partners.bank_ids\
+                    .filtered(lambda x: x.company_id in (False, wizard.company_id))._origin
+            else:
+                wizard.available_partner_bank_ids = False
 
-    @api.depends('partner_id')
+    @api.depends('available_partner_bank_ids')
     def _compute_partner_bank_id(self):
-        ''' The default partner_bank_id will be the first available on the partner. '''
         for wizard in self:
-            available_partner_bank_accounts = wizard.partner_id.bank_ids.filtered(lambda x: x.company_id in (False, wizard.company_id))
-            if available_partner_bank_accounts:
-                wizard.partner_bank_id = available_partner_bank_accounts[0]._origin
+            if wizard.can_edit_wizard:
+                if wizard.partner_bank_id:
+                    wizard.partner_bank_id = wizard.partner_bank_id
+                else:
+                    batches = wizard._get_batches()
+                    wizard.partner_bank_id = self.env['res.partner.bank'].browse(batches[0]['key_values']['partner_bank_id'])
```

Comment:
So we still set it if we different batches, with different accounts involved? Isn't it confusing?